### PR TITLE
Support for transpose attributes in MatMul operation

### PR DESCRIPTION
**Pull request by alexsu52**
### Changes

Support for transpose attributes in OpenVINO MatMul operation

### Reason for changes

Transpose attributes are not supported in OpenVINO MatMul operation. It leads to accuracy and performance degradation.

### Related tickets

ref: 111731

### Tests

TBD

### CODE REVIEW

File: nncf/openvino/graph/nncf_node_utils.py
Comment (l-bat) [inline nncf/openvino/graph/nncf_node_utils.py:1]: I think we can add `get_weight_channel_axis` to `nncf/openvino/graph/node_utils.py`.

File: nncf/openvino/graph/nncf_graph_builder.py [hunk 1]
```diff
@@ -177,9 +178,16 @@ def create_nncf_graph(model: ov.Model) -> NNCFGraph:
...
+                    transpose = False
+                    if metatype == OVMatMulMetatype:
+                        attribute_names = ["transpose_a", "transpose_b"]
+                        node_attributes = node.get_attributes()
+                        transpose = node_attributes[attribute_names[const_port_id]]
+
                     const_attrs[const_port_id] = {
                         "name": const_node.get_friendly_name(),
                         "shape": tuple(const_node.get_output_shape(0)),
+                        "transpose": transpose,
```
Comment (l-bat) [inline nncf/openvino/graph/nncf_graph_builder.py:190]: Can we add `transpose` attr only for `MatMul` ops?

File: nncf/openvino/graph/nncf_node_utils.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nncf.openvino.graph.metatypes.openvino_metatypes import GENERAL_WEIGHT_LAYER_METATYPES
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVMatMulMetatype
+from nncf.openvino.graph.nncf_graph_builder import OVConstantLayerAttributes
+
+
+def get_weight_channel_axis(node, weights_port_id):
```
Comment (l-bat) [inline nncf/openvino/graph/nncf_node_utils.py:17]:
```suggestion
def get_weight_channel_axis(node: NNCFNode, weights_port_id: int) -> int:
```

File: nncf/openvino/graph/nncf_node_utils.py
Comment (alexsu52) [reply nncf/openvino/graph/nncf_node_utils.py:1]: done

File: nncf/openvino/graph/nncf_graph_builder.py (hunk 1 above)
Comment (alexsu52) [reply nncf/openvino/graph/nncf_graph_builder.py:190]: done

File: nncf/openvino/graph/nncf_node_utils.py (hunk 2 above)
Comment (alexsu52) [reply nncf/openvino/graph/nncf_node_utils.py:17]: done

**Comment by AlexKoff88** [comment]
I suggest fixing the CI issues and merging this since it covers all the cases we are aware of at the moment

**Review by l-bat — APPROVED** [review]

**Comment by alexsu52** [comment]
run openvino pre-commit tests

**Pull request merged**